5.9.20220417.0: test suite is failing because it tries to use git

Despite fact that configure detested:
```console
checking building in a git repository... no
checking for git... git
```
test suite is trying to use git in some units
```console
+ cd ctags-p5.9.20220417.0
+ /usr/bin/make -O -j48 V=1 VERBOSE=1 check
rm -f optlib/cmake.c optlib/ctags-optlib.c optlib/elixir.c optlib/gdbinit.c optlib/inko.c optlib/iPythonCell.c optlib/kconfig.c optlib/lex.c optlib/man.c optlib/meson.c optlib/mesonOptions.c optlib/org.c optlib/passwd.c optlib/pod.c optlib/puppetManifest.c optlib/qemuhx.c optlib/rpmMacros.c optlib/scss.c optlib/systemtap.c optlib/yacc.c
/usr/bin/make optlib/cmake.c optlib/ctags-optlib.c optlib/elixir.c optlib/gdbinit.c optlib/inko.c optlib/iPythonCell.c optlib/kconfig.c optlib/lex.c optlib/man.c optlib/meson.c optlib/mesonOptions.c optlib/org.c optlib/passwd.c optlib/pod.c optlib/puppetManifest.c optlib/qemuhx.c optlib/rpmMacros.c optlib/scss.c optlib/systemtap.c optlib/yacc.c
make[1]: Entering directory '/home/tkloczko/rpmbuild/BUILD/ctags-p5.9.20220417.0'
./misc/optlib2c optlib/ctags-optlib.ctags > optlib/ctags-optlib.c
make[1]: Leaving directory '/home/tkloczko/rpmbuild/BUILD/ctags-p5.9.20220417.0'
make[1]: Entering directory '/home/tkloczko/rpmbuild/BUILD/ctags-p5.9.20220417.0'
./misc/optlib2c optlib/gdbinit.ctags > optlib/gdbinit.c
make[1]: Leaving directory '/home/tkloczko/rpmbuild/BUILD/ctags-p5.9.20220417.0'
make[1]: Entering directory '/home/tkloczko/rpmbuild/BUILD/ctags-p5.9.20220417.0'
./misc/optlib2c optlib/elixir.ctags > optlib/elixir.c
make[1]: Leaving directory '/home/tkloczko/rpmbuild/BUILD/ctags-p5.9.20220417.0'
make[1]: Entering directory '/home/tkloczko/rpmbuild/BUILD/ctags-p5.9.20220417.0'
./misc/optlib2c optlib/iPythonCell.ctags > optlib/iPythonCell.c
make[1]: Leaving directory '/home/tkloczko/rpmbuild/BUILD/ctags-p5.9.20220417.0'
make[1]: Entering directory '/home/tkloczko/rpmbuild/BUILD/ctags-p5.9.20220417.0'
./misc/optlib2c optlib/cmake.ctags > optlib/cmake.c
make[1]: Leaving directory '/home/tkloczko/rpmbuild/BUILD/ctags-p5.9.20220417.0'
make[1]: Entering directory '/home/tkloczko/rpmbuild/BUILD/ctags-p5.9.20220417.0'
./misc/optlib2c optlib/kconfig.ctags > optlib/kconfig.c
make[1]: Leaving directory '/home/tkloczko/rpmbuild/BUILD/ctags-p5.9.20220417.0'
make[1]: Entering directory '/home/tkloczko/rpmbuild/BUILD/ctags-p5.9.20220417.0'
./misc/optlib2c optlib/passwd.ctags > optlib/passwd.c
make[1]: Leaving directory '/home/tkloczko/rpmbuild/BUILD/ctags-p5.9.20220417.0'
make[1]: Entering directory '/home/tkloczko/rpmbuild/BUILD/ctags-p5.9.20220417.0'
./misc/optlib2c optlib/man.ctags > optlib/man.c
make[1]: Leaving directory '/home/tkloczko/rpmbuild/BUILD/ctags-p5.9.20220417.0'
make[1]: Entering directory '/home/tkloczko/rpmbuild/BUILD/ctags-p5.9.20220417.0'
./misc/optlib2c optlib/lex.ctags > optlib/lex.c
make[1]: Leaving directory '/home/tkloczko/rpmbuild/BUILD/ctags-p5.9.20220417.0'
make[1]: Entering directory '/home/tkloczko/rpmbuild/BUILD/ctags-p5.9.20220417.0'
./misc/optlib2c optlib/mesonOptions.ctags > optlib/mesonOptions.c
make[1]: Leaving directory '/home/tkloczko/rpmbuild/BUILD/ctags-p5.9.20220417.0'
make[1]: Entering directory '/home/tkloczko/rpmbuild/BUILD/ctags-p5.9.20220417.0'
./misc/optlib2c optlib/inko.ctags > optlib/inko.c
make[1]: Leaving directory '/home/tkloczko/rpmbuild/BUILD/ctags-p5.9.20220417.0'
make[1]: Entering directory '/home/tkloczko/rpmbuild/BUILD/ctags-p5.9.20220417.0'
./misc/optlib2c optlib/meson.ctags > optlib/meson.c
make[1]: Leaving directory '/home/tkloczko/rpmbuild/BUILD/ctags-p5.9.20220417.0'
make[1]: Entering directory '/home/tkloczko/rpmbuild/BUILD/ctags-p5.9.20220417.0'
./misc/optlib2c optlib/rpmMacros.ctags > optlib/rpmMacros.c
make[1]: Leaving directory '/home/tkloczko/rpmbuild/BUILD/ctags-p5.9.20220417.0'
make[1]: Entering directory '/home/tkloczko/rpmbuild/BUILD/ctags-p5.9.20220417.0'
./misc/optlib2c optlib/org.ctags > optlib/org.c
make[1]: Leaving directory '/home/tkloczko/rpmbuild/BUILD/ctags-p5.9.20220417.0'
make[1]: Entering directory '/home/tkloczko/rpmbuild/BUILD/ctags-p5.9.20220417.0'
./misc/optlib2c optlib/pod.ctags > optlib/pod.c
make[1]: Leaving directory '/home/tkloczko/rpmbuild/BUILD/ctags-p5.9.20220417.0'
\
builddir=$(pwd); \
mkdir -p ${builddir}/misc; \
if test -s './mini-geany'; then \
        if /bin/sh ./misc/tlib ./mini-geany \
                ./misc/mini-geany.expected \
                ${builddir}/misc/mini-geany.actual \
                ; then \
                echo 'mini-geany: OK'; \
        else \
                echo 'mini-geany: FAILED'; \
        fi; \
else \
        echo 'mini-geany: SKIP'; \
fi
 ./mini-geany output:
This parser only parses C files - provide them as arguments on the command line or get a hard-coded buffer parsed when no arguments are provided

Number of all parsers is: 148
We are parsing C which provides the following kinds:
d: macro
e: enumerator
f: function
g: enum
h: header
l: local
m: member
p: prototype
s: struct
t: typedef
u: union
v: variable
x: externvar
z: parameter
L: label
D: macroparam

Parsing buffer:
foo     line: 1 kind: function   lang: C
bar     line: 3 kind: function   lang: C
main    line: 5 kind: function   lang: C
comparing with expected output:
mini-geany: OK
make[1]: Entering directory '/home/tkloczko/rpmbuild/BUILD/ctags-p5.9.20220417.0'
./misc/optlib2c optlib/qemuhx.ctags > optlib/qemuhx.c
make[1]: Leaving directory '/home/tkloczko/rpmbuild/BUILD/ctags-p5.9.20220417.0'
make[1]: Entering directory '/home/tkloczko/rpmbuild/BUILD/ctags-p5.9.20220417.0'
./misc/optlib2c optlib/scss.ctags > optlib/scss.c
make[1]: Leaving directory '/home/tkloczko/rpmbuild/BUILD/ctags-p5.9.20220417.0'
make[1]: Entering directory '/home/tkloczko/rpmbuild/BUILD/ctags-p5.9.20220417.0'
./misc/optlib2c optlib/systemtap.ctags > optlib/systemtap.c
make[1]: Leaving directory '/home/tkloczko/rpmbuild/BUILD/ctags-p5.9.20220417.0'
make[1]: Entering directory '/home/tkloczko/rpmbuild/BUILD/ctags-p5.9.20220417.0'
./misc/optlib2c optlib/yacc.ctags > optlib/yacc.c
make[1]: Leaving directory '/home/tkloczko/rpmbuild/BUILD/ctags-p5.9.20220417.0'
make[1]: Entering directory '/home/tkloczko/rpmbuild/BUILD/ctags-p5.9.20220417.0'
./misc/optlib2c optlib/puppetManifest.ctags > optlib/puppetManifest.c
make[1]: Leaving directory '/home/tkloczko/rpmbuild/BUILD/ctags-p5.9.20220417.0'
if ! git diff --exit-code optlib; then \
        echo "Files under optlib are not up to date." ; \
        echo "If you change optlib/foo.ctags, don't forget to add optlib/foo.c to your commit." ; \
        exit 1 ; \
else \
        echo "Files under optlib are up to date." ; \
fi
warning: Not a git repository. Use --no-index to compare two paths outside a working tree
usage: git diff --no-index [<options>] <path> <path>

Diff output format options
    -p, --patch           generate patch
    -s, --no-patch        suppress diff output
    -u                    generate patch
    -U, --unified[=<n>]   generate diffs with <n> lines context
    -W, --function-context
                          generate diffs with <n> lines context
    --raw                 generate the diff in raw format
    --patch-with-raw      synonym for '-p --raw'
    --patch-with-stat     synonym for '-p --stat'
    --numstat             machine friendly --stat
    --shortstat           output only the last line of --stat
    -X, --dirstat[=<param1,param2>...]
                          output the distribution of relative amount of changes for each sub-directory
    --cumulative          synonym for --dirstat=cumulative
    --dirstat-by-file[=<param1,param2>...]
                          synonym for --dirstat=files,param1,param2...
    --check               warn if changes introduce conflict markers or whitespace errors
    --summary             condensed summary such as creations, renames and mode changes
    --name-only           show only names of changed files
    --name-status         show only names and status of changed files
    --stat[=<width>[,<name-width>[,<count>]]]
                          generate diffstat
    --stat-width <width>  generate diffstat with a given width
    --stat-name-width <width>
                          generate diffstat with a given name width
    --stat-graph-width <width>
                          generate diffstat with a given graph width
    --stat-count <count>  generate diffstat with limited lines
    --compact-summary     generate compact summary in diffstat
    --binary              output a binary diff that can be applied
    --full-index          show full pre- and post-image object names on the "index" lines
    --color[=<when>]      show colored diff
    --ws-error-highlight <kind>
                          highlight whitespace errors in the 'context', 'old' or 'new' lines in the diff
    -z                    do not munge pathnames and use NULs as output field terminators in --raw or --numstat
    --abbrev[=<n>]        use <n> digits to display object names
    --src-prefix <prefix>
                          show the given source prefix instead of "a/"
    --dst-prefix <prefix>
                          show the given destination prefix instead of "b/"
    --line-prefix <prefix>
                          prepend an additional prefix to every line of output
    --no-prefix           do not show any source or destination prefix
    --inter-hunk-context <n>
                          show context between diff hunks up to the specified number of lines
    --output-indicator-new <char>
                          specify the character to indicate a new line instead of '+'
    --output-indicator-old <char>
                          specify the character to indicate an old line instead of '-'
    --output-indicator-context <char>
                          specify the character to indicate a context instead of ' '

Diff rename options
    -B, --break-rewrites[=<n>[/<m>]]
                          break complete rewrite changes into pairs of delete and create
    -M, --find-renames[=<n>]
                          detect renames
    -D, --irreversible-delete
                          omit the preimage for deletes
    -C, --find-copies[=<n>]
                          detect copies
    --find-copies-harder  use unmodified files as source to find copies
    --no-renames          disable rename detection
    --rename-empty        use empty blobs as rename source
    --follow              continue listing the history of a file beyond renames
    -l <n>                prevent rename/copy detection if the number of rename/copy targets exceeds given limit

Diff algorithm options
    --minimal             produce the smallest possible diff
    -w, --ignore-all-space
                          ignore whitespace when comparing lines
    -b, --ignore-space-change
                          ignore changes in amount of whitespace
    --ignore-space-at-eol
                          ignore changes in whitespace at EOL
    --ignore-cr-at-eol    ignore carrier-return at the end of line
    --ignore-blank-lines  ignore changes whose lines are all blank
    -I, --ignore-matching-lines <regex>
                          ignore changes whose all lines match <regex>
    --indent-heuristic    heuristic to shift diff hunk boundaries for easy reading
    --patience            generate diff using the "patience diff" algorithm
    --histogram           generate diff using the "histogram diff" algorithm
    --diff-algorithm <algorithm>
                          choose a diff algorithm
    --anchored <text>     generate diff using the "anchored diff" algorithm
    --word-diff[=<mode>]  show word diff, using <mode> to delimit changed words
    --word-diff-regex <regex>
                          use <regex> to decide what a word is
    --color-words[=<regex>]
                          equivalent to --word-diff=color --word-diff-regex=<regex>
    --color-moved[=<mode>]
                          moved lines of code are colored differently
    --color-moved-ws <mode>
                          how white spaces are ignored in --color-moved

Other diff options
    --relative[=<prefix>]
                          when run from subdir, exclude changes outside and show relative paths
    -a, --text            treat all files as text
    -R                    swap two inputs, reverse the diff
    --exit-code           exit with 1 if there were differences, 0 otherwise
    --quiet               disable all output of the program
    --ext-diff            allow an external diff helper to be executed
    --textconv            run external text conversion filters when comparing binary files
    --ignore-submodules[=<when>]
                          ignore changes to submodules in the diff generation
    --submodule[=<format>]
                          specify how differences in submodules are shown
    --ita-invisible-in-index
                          hide 'git add -N' entries from the index
    --ita-visible-in-index
                          treat 'git add -N' entries as real in the index
    -S <string>           look for differences that change the number of occurrences of the specified string
    -G <regex>            look for differences that change the number of occurrences of the specified regex
    --pickaxe-all         show all changes in the changeset with -S or -G
    --pickaxe-regex       treat <string> in -S as extended POSIX regular expression
    -O <file>             control the order in which files appear in the output
    --rotate-to <path>    show the change in the specified path first
    --skip-to <path>      skip the output to the specified path
    --find-object <object-id>
                          look for differences that change the number of occurrences of the specified object
    --diff-filter [(A|C|D|M|R|T|U|X|B)...[*]]
                          select files by diff type
    --output <file>       Output to a specific file

Files under optlib are not up to date.
If you change optlib/foo.ctags, don't forget to add optlib/foo.c to your commit.
make: *** [Makefile:7420: check-genfile] Error 1
make: *** Waiting for unfinished jobs....

/usr/bin/gcc -std=gnu99 -Wall   -I/usr/include/libxml2      -O2 -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fdata-sections -ffunction-sections -flto=auto -flto-partition=none  -Wl,-z,relro -Wl,--as-needed -Wl,--gc-sections -Wl,-z,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -flto=auto -flto-partition=none -fuse-linker-plugin -Wl,--build-id=sha1 -o ctags  main/ctags-cmd.o libctags.a gnulib/libgnu.a  -lxml2  -ljansson  -lyaml  -lseccomp   -lpcre2-8
```

--- a/circle.yml
+++ b/circle.yml
@@ -160,7 +160,15 @@ jobs:
            name: Test
            command: |
              ulimit -c 0
-             make distcheck
+             # Let's trick git.
+             #
+             # "git" running in "make distcheck" may refer $(pwd)/.git of THIS context.
+             # On the other hand, we want to run the test cases in an environment separated
+             # from our git repo.
+             #
+             mkdir __NO_GIT__
+             make distcheck GIT_DIR=__NO_GIT__
+             rmdir __NO_GIT__
 
    ubi8_make:
      working_directory: ~/universal-ctags

--- a/makefiles/testing.mak
+++ b/makefiles/testing.mak
@@ -296,6 +296,7 @@ chkgen_verbose = $(chkgen_verbose_@AM_V@)
 chkgen_verbose_ = $(chkgen_verbose_@AM_DEFAULT_V@)
 chkgen_verbose_0 = @echo CHKGEN "    $@";
 check-genfile:
+if BUILD_IN_GIT_REPO
 # OPTLIB2C_SRCS : committed for win32 build
 	$(chkgen_verbose)rm -f $(OPTLIB2C_SRCS)
 	$(chkgen_verbose)$(MAKE) $(OPTLIB2C_SRCS)
@@ -351,3 +352,4 @@ endif
 	else \
 		echo "Files under win32 are up to date." ; \
 	fi
+endif

